### PR TITLE
MVP-181 DevHub publishing pull request

### PR DESCRIPTION
**Pull request by TaylorFries**
## 🎯 Summary

  
[MVP-181](https://citz-imb.atlassian.net/browse/MVP-181)

Added bug report and feature request issue templates
This pull request will add the files required to publish documentation on DevHub

## 🔰 Checklist

- [x] I have read and agree with the following checklist and am following the guidelines in our [Code of Conduct](/CODE_OF_CONDUCT.md) document.

> - I have performed a self-review of my code.
> - I have commented my code, particularly in hard-to-understand areas.
> - I have made corresponding changes to the documentation where required.
> - I have tested my changes to the best of my ability.
> - I have consulted with the team if introducing a new dependency.
> - My changes generate no new warnings.

### CODE REVIEW

**Review by SDToews — DISMISSED** [review]

File: techdocs/docs/using-the-package/protecting-a-route.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+// Users must have EITHER 'Member' or 'Verified' role.
+app.use("/vote", protectedRoute(['Member', 'Verified'], { requireAllRoles: false }), voteRouter);
+```
+
+!!! note "Note
+    By default, all roles in the array will be required.  
+    If you wish to require only one of a list of roles, add the options parameter with `requireAllRoles` property set to `false`.
+
+!!! tip "Tip"
+    You can also check if the user has a role within the controller function by importing and using the `hasRoles` function.
```
Comment (BradyMitch) [inline techdocs/docs/using-the-package/protecting-a-route.md:57]:
```suggestion
    You can also check if the user has a role within the controller function by using the `hasRoles` function on the `SSOUser` object of `req.user`.
```
Changed from an imported function to a property of the user object and this doc never changed.

File: techdocs/pubcode.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+version: 1
+data_management_roles:
+  product_owner: 
+product_information:
+  ministry: CITZ
+  product_acronym: 
+  product_description: An npm package that offers an integration solution for connecting Express applications requiring authentication to the B.C. government's Single Sign-On service.
+  product_name: CITZ IMB Express (NPM Package)
```
Comment (BradyMitch) [inline techdocs/pubcode.yaml:8]:
```suggestion
  product_name: CITZ IMB SSO Express (NPM Package)
```

File: catalog-info.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+apiVersion: backstage.io/v1alpha1
+kind: Component
+metadata:
+  name: citz-imb-sso-express-(npm-package)
```
Comment (BradyMitch) [inline catalog-info.yaml:4]: In the sso react package we dont have the brackets here. I think we should keep them the same.

File: catalog-info.yaml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  title: CITZ IMB SSO Express (NPM Package)
+  description: An npm package that offers an integration solution for connecting
+    Express applications requiring authentication to the B.C. government's
+    Single Sign-On service.
+  annotations:
+    github.com/project-slug: bcgov/citz-imb-sso-express
+    backstage.io/techdocs-ref: dir:.
+    bcgovpubcode.gov.bc.ca/ministry: CITZ
+    bcgovpubcode.gov.bc.ca/product_name: CITZ IMB Express (NPM Package)
```
Comment (BradyMitch) [inline catalog-info.yaml:13]: In the sso react package this is just `citz-imb-sso-react`

**Review by BradyMitch — COMMENTED** [review]
We should be sure to remove the old assets directory after these are live in both sso-react and here.

File: techdocs/pubcode.yaml (hunk 2 above)
Comment (BradyMitch) [reply techdocs/pubcode.yaml:8]: Actually in the sso react package this is just `citz-imb-sso-react`

File: techdocs/docs/getting-started/quick-start-guide.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
+# Quick Start Guide
+
+Set up in 2 steps. 
+
+### 1. Initialize with Express App 
+
+Add import for `sso` to the file that defines the express app.  
+Add `sso(app);` below the definition of the express app, where `app` is defined by `express()`.
+
+#### Import Examples:
```
Comment (BradyMitch) [inline techdocs/docs/getting-started/quick-start-guide.md:10]:
```suggestion
#### `Import Examples:`
```
Any time we use `Example` in the sso react docs I used the backticks format.

File: techdocs/docs/getting-started/quick-start-guide.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+```JavaScript
+// ESModule import (preferred).
+import { sso } from '@bcgov/citz-imb-sso-express';
+
+// CommonJS import.
+const { sso } = require('@bcgov/citz-imb-sso-express');
+```
+
+#### Use Example:
```
Comment (BradyMitch) [inline techdocs/docs/getting-started/quick-start-guide.md:20]:
```suggestion
#### `Use Example:`
```

File: techdocs/docs/using-the-package/init-options.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
+# Initialization Options
+
+Add additional functionality to your SSO integration by using the optional second parameter to the `sso()` function which is of type `SSOOptions`.  
+Currently `afterUserLogin` and `afterUserLogout` are available to configure. 
+
+### Example:
```
Comment (BradyMitch) [inline techdocs/docs/using-the-package/init-options.md:6]:
```suggestion
### `Example:`
```

File: techdocs/README_TECHDOCS.md [hunk 8]
```diff
@@ -0,0 +1,123 @@
+[![Package Status](https://github.com/bcgov/citz-imb-sso-express/actions/workflows/publish.yaml/badge.svg)](https://github.com/bcgov/citz-imb-sso-express/actions/workflows/publish.yaml)
+
+
+# citz-imb-sso-express-(npm-package)
```
[the remark this answers is not in the document]
Comment (TaylorFries) [reply techdocs/README_TECHDOCS.md]: newest commit adds these changes

File: mkdocs.yml [hunk 9]
```diff
@@ -0,0 +1,19 @@
+site_name: "CITZ IMB SSO Express (NPM Package)"
+site_description: "An npm package that offers an integration solution for connecting Express applications requiring authentication to the B.C. government's Single Sign-On service."
+docs_dir: "docs"
```
Comment (BradyMitch) [inline mkdocs.yml:3]:
```suggestion
docs_dir: "techdocs/docs"
```

Comment (TaylorFries) [reply mkdocs.yml]: added in newest commit

File: techdocs/README_TECHDOCS.md [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+Note that MkDocs uses `mkdocs.yml`, not `mkdocs.yaml`, although both appear to work.
+Refer to the [mkdocs user guide](https://www.mkdocs.org/user-guide/configuration/) for more details.
+
+## Previewing documentation locally
+
+You will likely want to preview your documentation on your computer. Follow [these instructions](https://github.com/bcgov/devhub-techdocs-publish/blob/main/docs/index.md#how-to-use-the-docker-image-to-preview-content-locally) to do so.
+
+## Building and publishing your documentation
+
+The [GitHub Action workflow file](.github/workflows/publish.yaml) can validate, build (produce browser-friendly HTML from Markdown) and publish your documentation to DevHub automatically.  Initially, it is set up just to validate and build your documentation.
```
Comment (BradyMitch) [inline techdocs/README_TECHDOCS.md:89]: This should be updated

File: techdocs/README_TECHDOCS.md [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+
+## Building and publishing your documentation
+
+The [GitHub Action workflow file](.github/workflows/publish.yaml) can validate, build (produce browser-friendly HTML from Markdown) and publish your documentation to DevHub automatically.  Initially, it is set up just to validate and build your documentation. 
+
+The portion of the workflow file that publishes your documentation is commented out because it requires addition set up with the Developer Experience team. When you'd like to have your documentation published in DevHub, [contact the Developer Experience Team](mailto:developer.experience@gov.bc.ca) who will work with you to set things up. 
+
+### The publishing workflow
+
+DevHub has a "preview" environment and a production environment, and once the Developer Experience team has you set up, and you've uncommented the publishing sections in the [workflow file](.github/workflows/publish.yaml) you'll be in control of how and when your content is published to each. You'll be able to control this by modifying the [workflow file](.github/workflows/publish.yaml) to suit your process and team.
```
Comment (BradyMitch) [inline techdocs/README_TECHDOCS.md:95]: Same here

File: techdocs/docs/index.md [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+This npm package offers an integration solution for Express applications requiring authentication through the B.C. government's Single Sign-On ([CSS]) service. It abstracts the complexity of handling SSO protocols manually. By using this package, developers can quickly implement authentication and authorization in their Express applications to meet B.C. government security standards.
+
+- Built for a NodeJS:20 Express API.
+- For [CSS] Gold Standard with usecase set as `Browser Login and Service Account`.
+- Works with Vanilla JavaScript or TypeScript 5.
+- For use with a react application using [@bcgov/citz-imb-sso-react]
+
+<br />
+
+!!! info "Important"
```
Comment (BradyMitch) [inline techdocs/docs/index.md:14]:
```suggestion
!!! warn "Important"
```

File: techdocs/docs/purpose-and-benefits.md [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+
+## Overview
+
+The [CITZ IMB Common Code] team enhances efficiency and maintainability of our React/Express applications by identifying reusable code within our products, packaging it, and adding it to the npm registry for our teams to utilize.
+
+We created these packages to integrate with the Common Hosted Single Sign-On ([CSS]) Service provided by the Office of the Chief Information Officer.
+
+<br />
+
+!!! info "Important"
```
Comment (BradyMitch) [inline techdocs/docs/purpose-and-benefits.md:11]:
```suggestion
!!! warning "Important"
```

File: techdocs/docs/using-the-package/getting-user-information.md [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+
+// Users must have EITHER 'Member' or 'Verified' role.
+if (user?.hasRoles(['Member', 'Verified'], { requireAllRoles: false })) // Do Something...
+```
+
+!!! note "Note"
+    By default, all roles in the array will be required.  
+    If you wish to require only one of a list of roles, add the options parameter with `requireAllRoles` property set to `false`.
+
+!!! warn "Important"
```
Comment (BradyMitch) [inline techdocs/docs/using-the-package/getting-user-information.md:27]:
```suggestion
!!! warning "Important"
```
Not sure if warn works but you had it as warning on the other repo docs

File: techdocs/pubcode.yaml [hunk 15]
```diff
@@ -0,0 +1,10 @@
+version: 1
+data_management_roles:
+  product_owner: 
+product_information:
+  ministry: CITZ
+  product_acronym: 
+  product_description: An npm package that offers an integration solution for connecting Express applications requiring authentication to the B.C. government's Single Sign-On service.
+  product_name: CITZ IMB SSO Express (NPM Package)
```
Comment (BradyMitch) [inline techdocs/pubcode.yaml:8]:
```suggestion
  product_name: citz-imb-sso-express
```
Match what it is in react package

File: techdocs/README_TECHDOCS.md [hunk 16]
```diff
@@ -0,0 +1,120 @@
...
+* A release in your repository will publish the code to the [production DevHub website](http://developer.gov.bc.ca).
+
+## Getting help or reporting an issue
+
+To report bugs/issues/features requests, please email the Developer Experience team at [Developer.Experience@gov.bc.ca](mailto:Developer.Experience@gov.bc.ca)
+
+The full documentation for MkDocs (which is powering the DevHub TechDocs function) is at the [MkDocs website](https://www.mkdocs.org/user-guide/writing-your-docs/). Note that not all content on the site will be applicable to DevHub's use of MkDocs.
+
+## Contributing
+If you would like to contribute, please see our [contributing](CONTRIBUTING.md) guidelines.
```
Comment (BradyMitch) [inline techdocs/README_TECHDOCS.md:109]: We should update this sentence because we do not have a contributing document

Comment (TaylorFries) [reply techdocs/README_TECHDOCS.md]: updated in newest commit - [f74d835](https://github.com/bcgov/citz-imb-sso-express/pull/70/commits/f74d835b112e0adb23c3461aa43e6929f6802bbc)

File: techdocs/README_TECHDOCS.md (hunk 11 above)
Comment (TaylorFries) [reply techdocs/README_TECHDOCS.md]: updated in newest commit - [f74d835](https://github.com/bcgov/citz-imb-sso-express/pull/70/commits/f74d835b112e0adb23c3461aa43e6929f6802bbc)

File: techdocs/README_TECHDOCS.md (hunk 10 above)
Comment (TaylorFries) [reply techdocs/README_TECHDOCS.md]: updated in newest commit - [f74d835](https://github.com/bcgov/citz-imb-sso-express/pull/70/commits/f74d835b112e0adb23c3461aa43e6929f6802bbc)

**Review by BradyMitch — APPROVED** [review]

**Pull request merged**